File: v2/src/main/java/com/api/v2/customers/domain/CustomerRepository.java
package com.api.v2.customers.domain;

import com.api.v2.persons.domain.Person;
import org.springframework.data.mongodb.repository.Query;
import org.springframework.data.mongodb.repository.ReactiveMongoRepository;
import reactor.core.publisher.Mono;

import java.util.UUID;

public interface CustomerRepository extends ReactiveMongoRepository<Customer, UUID> {

    @Query("{ 'person': ?0 }")
    Mono<Customer> findByPerson(Person person);
}

File: kotlin/v3/src/main/kotlin/com/api/v3/customers/CustomerRepository.kt
package com.api.v3.customers

import com.api.v3.persons.domain.Person
import org.springframework.data.mongodb.repository.Query
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import java.util.UUID

interface CustomerRepository: CoroutineCrudRepository<Customer, UUID> {

    @Query("{ 'person': ?0 }")
    fun findByPerson(person: Person): Person

}